code/lib/core-server/src/utils/StoryIndexGenerator.ts

```diff
@@ -326,7 +330,12 @@ export class StoryIndexGenerator {
         name,
         importPath,
         type: 'docs',
-        tags: [...metaTags, 'docs', ...(!hasAutodocsTag && !isStoriesMdx ? [AUTODOCS_TAG] : [])],
+        metaTags,
+        tags: combineTags(
+          ...metaTags,
+          'docs',
+          ...(!hasAutodocsTag && !isStoriesMdx ? [AUTODOCS_TAG] : [])
```

Comment:
Could we make `combineTags` drop duplicates so we don't need this check for `hasAutodocsTag`?